File: Java/ID3.java
/*
ID: 3
The prime factors of 13195 are 5, 7, 13 and 29. What is the largest prime factor of the number 600851475143?
*/
class ID3 {
	public static long MaxPrimeFactorOf(long n) {
		long max = 0L;
		for (long i = 2L; i <= n; i++) {
			for (long l = 1L; l <= Math.sqrt(i); l++) {
				if (l % i == 0) {
					break;
				} else {
					while (n % i == 0) {
						n = n / i;
						max = i;
					}
				}
			}
		}
		return max;
	}

	public static void main(String[] args) {
		System.out.println(MaxPrimeFactorOf(600851475143L));
	}
}
// Answer: 6857

File: R/ID3.R
#ID: 3
#The prime factors of 13195 are 5, 7, 13 and 29. What is the largest prime factor of the number 600851475143?

is.prime <- function(x) { 
  if (x == 1) {
    flag <- FALSE
  } 
  if (x == 2 | x == 3) {
    flag <-  TRUE
  } 
  else {
    flag <- !any(x %% 2:(floor(sqrt(x))) == 0)
  }
  return(flag)
}

max.prime.factor.of <- function(num) {
n <- floor(sqrt(num))
flag <- logical(n)
for (i in 1:n) {
  flag[i] <- is.prime(i)
}
prime <- (1:n)[flag]
prime.fac <- prime[(num %% prime) == 0]
answer <- max(prime.fac)
return(answer)
}
max.prime.factor.of(600851475143)
#Answer:  6857
